src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -0,0 +1,424 @@
+            }
+
+            int retVal = libssl.SSL_do_handshake(context.sslPtr);
+            if ((retVal == 1) && !isServer)
+            {            
+                return true;
+            }
+
+            int error;
+            if (retVal != 1)
+            {
+                error = GetSslError(context.sslPtr, retVal, "SSL Handshake failed: ");
+                if ((retVal != -1) || (error != libssl.SslErrorCode.SSL_ERROR_WANT_READ))
+                {
+                    throw CreateSslException(context.sslPtr, "SSL Handshake failed: ", error);
+                }
+            }
+
+            sendCount = libssl.BIO_ctrl_pending(context.writeBioPtr);
+
+            if (sendCount > 0)
+            {
+                sendPtr = Marshal.AllocHGlobal(sendCount);
+                sendCount = BioRead(context.writeBioPtr, sendPtr, sendCount);
+                if (sendCount <= 0)
+                {
+                    error = sendCount;
+                    Marshal.FreeHGlobal(sendPtr);
+                    sendPtr = IntPtr.Zero;
+                    sendCount = 0;
+                    error = GetSslError(context.sslPtr, error, "Read Bio failed: ");                 
+                    throw CreateSslException("SSL Handshake failed", error);                   
+                }
+            }
+        
+            return ((libssl.SSL_state(context.sslPtr) == libssl.SslState.SSL_ST_OK)) ? true : false;
+
+        }
+
+        internal static int Encrypt(IntPtr handlePtr, IntPtr buffer, int count, int bufferCapacity)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(handlePtr);  
+
+            var retVal = libssl.SSL_write(context.sslPtr, buffer, count);
+
+            if (retVal != count)
+            {
+                throw CreateSslException("OpenSsl::Encrypt failed");
+            }
+
+            int capacityNeeded = libssl.BIO_ctrl_pending(context.writeBioPtr);      
+
+            if (retVal == count)
+            {
+                if (capacityNeeded > bufferCapacity)
+                {
+                    throw CreateSslException("OpenSsl::Encrypt outBufferPtr should be null");
+                }
+
+                IntPtr outBufferPtr = buffer;
+
+                retVal = BioRead(context.writeBioPtr, outBufferPtr, capacityNeeded);
+
+                if (retVal < 0)
+                {
+                    throw CreateSslException("OpenSsl::Encrypt failed");
+                }
+            }
+
+            return retVal;
+        }
+
+        internal static int Decrypt(IntPtr sslContextPtr, IntPtr outBufferPtr, int count)
+        {
+            SslContext context = Marshal.PtrToStructure<SslContext>(sslContextPtr);
+
+            var retVal = BioWrite(context.readBioPtr, outBufferPtr, count);
+
+            if (retVal == count)
+            {
+                retVal = libssl.SSL_read(context.sslPtr, outBufferPtr, retVal);
+
+                if (retVal > 0) count = retVal;
```

Comment:
Style: Unbraced ifs require the body to be on a newline.
Style: Unbraced ifs are not allowed within a braced if.